NMS-14475: Update OIA to OpenNMS Plugin API (#5152)
OpenNMS/opennms: docs/modules/releasenotes/pages/whatsnew.adoc

== Breaking Changes

=== Running as non-root

> By default, OpenNMS now runs as the `opennms` user rather than `root`.

Reviewer: ```suggestion
By default, {page-component-title} now runs as the `opennms` user rather than `root`.
```

WARNING: If you are upgrading from a version of OpenNMS Horizon older than 29, you will have to fix the ownership of your files.

==== Fixing permissions and ownership

> Since many OpenNMS instances are very large with thousands or even millions of RRD and similar data files in `$OPENNMS_HOME/share`,

Reviewer: ```suggestion
Since many {page-component-title} instances are very large with thousands or even millions of RRD and similar data files in `$\{OPENNMS_HOME}/share`,
```

----

> Save the file and reload systemd with `systemctl daemon-reload` and restart OpenNMS with `systemctl restart opennms`.

Reviewer: ```suggestion
Save the file and reload systemd with `systemctl daemon-reload` and restart {page-component-title} with `systemctl restart opennms`.
```

> The OpenNMS process should run as the `root` user instead of `opennms` and can be verified with `ps -aux | grep java`.

Reviewer: ```suggestion
The {page-component-title} process should run as the `root` user instead of `opennms` and can be verified with `ps -aux | grep java`.
```

==== Trapd port changes

> Because OpenNMS can no longer listen on privileged ports by default, the default trapd port is now `10162`.

Reviewer: ```suggestion
Because {page-component-title} can no longer listen on privileged ports by default, the default trapd port is now `10162`.
```

You will need to configure a proxy agent or port forward to continue to receive traps at port `162`.
Otherwise, you must configure your agents to send traps to the new port instead.

==== Java ICMP permissions

> OpenNMS will attempt to configure ICMP permissions using the `net.ipv4.ping_group_range` sysctl.

Reviewer: ```suggestion
{page-component-title} will attempt to configure ICMP permissions using the `net.ipv4.ping_group_range` sysctl.
```

However, Linux kernels older than version 3.11 (like the version provided by CentOS 7) do not support this setting fully.

If you are on a distribution with an older unsupported kernel, you can give Java ICMP permissions using the `setcap` utility once you have completed your upgrade.

> Note that this example assumes you have already run `$OPENNMS_HOME/bin/install` after installing or upgrading, so `$OPENNMS_HOME/etc/java.conf` exists.

Reviewer: ```suggestion
Note that this example assumes you have already run `$\{OPENNMS_HOME}/bin/install` after installing or upgrading, so `$\{OPENNMS_HOME}/etc/java.conf` exists.
```

=== Additional configuration needed when using Kafka/gRPC for Minion communication.

The Twin API has replaced the OpenNMS REST API for synchronizing trapd configuration updates.
Additional settings are needed on both Minion and {page-component-title} to enable Twin when not using the ActiveMQ message broker.

Refer to the Deployment section in the documentation for details.

NOTE: Horizon 29.0.5 consolidated IPC features into one feature on Minion.
This groups all IPC (Sink/RPC/Twin) features of JMS into one feature as `openms-core-ipc-jms`.
Similarly for Kafka and gRPC.
When using Kafka or gRPC, you need to disable `opennms-core-ipc-jms` instead of disabling individual features such as `opennms-core-ipc-sink-camel`.

=== SQS Minion IPC support dropped

Support for Amazon SQS for Minion communication has been removed.

> We recommend to use Kafka instead for use cases that require horizontal scaling of Minion communication.

Reviewer 2: ```suggestion
We recommend that you use Kafka instead for use cases that require horizontal scaling of Minion communication.
```

> == Important Internal Changes

Author: ```suggestion
== Important internal changes
```
